Library/Homebrew/abstract_command.rb

```diff
@@ -32,6 +32,12 @@ def command_name
              .delete_suffix("-cmd")
       end
 
+      sig { void }
+      def raise_sh_command_error!
+        raise StandardError, "This command is just here for completions generation. " \
+                             "It's actually defined in `cmd/#{command_name}` instead."
```

Comment:
You could probably put this into the default definition of `#run` so you can just leave `#run` unimplemented for Ruby stubs.

Reply:
I'd say being more explicit here seems better; to me it doesn't make much sense to stub by default (which also presumes the `.sh` counterpart exists).

Reply:
How about:

```rb
# central file
module ShellCommand
  def run
    raise ...
  end
end

# in each cmd
class Update < AbstractCommand
  include ShellCommand

  ...
end
```

Bonus is it works as a metadata marker and could mean we have a test that validates this: e.g. `update.sh` must exist if `Homebrew::Cmd::Update.include?(ShellCommand)`

Reply:
Can we do this as a follow-up refactoring instead? This PR is an improvement as-is.